### ISSUE TITLE
Request Money - On entering emoji in merchant input field the height changes

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#26735]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open FAB Menu > Request Money > Next > Cash
2. Click Show more option > Merchant

## Expected Result:
The input field's height should remain consistent after entering an emoji

## Actual Result:
The input field's height changes

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.63.0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

![image (30)](https://github.com/Expensify/App/assets/93399543/fb86e8a1-c825-4900-825d-50851a453488)

https://github.com/Expensify/App/assets/93399543/9e822707-29b2-437d-90d9-8248afd6e145

**Expensify/Expensify Issue URL:**
**Issue reported by:** @aman-atg
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1693481505877419

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by alphaboss1104** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.

Request Money - On entering emoji in merchant input field the height changes

### What is the root cause of that problem?

The root cause of this issue is we didn't add `autoGrowHeight` props in `TextInput` component here.

https://github.com/Expensify/App/blob/8150ce011bcc4c685cb08627681a6fb1fe4276c7/src/pages/iou/MoneyRequestMerchantPage.js#L110-L119

### What changes do you think we should make in order to solve the problem?

We should only add `autoGrowHeight` props in [here](https://github.com/Expensify/App/blob/8150ce011bcc4c685cb08627681a6fb1fe4276c7/src/pages/iou/MoneyRequestMerchantPage.js#L110)

```js
              <TextInput
                  inputID="moneyRequestMerchant"
                  name="moneyRequestMerchant"
                  defaultValue={iou.merchant}
                  maxLength={CONST.MERCHANT_NAME_MAX_LENGTH}
                  label={translate('common.merchant')}
                  accessibilityLabel={translate('common.merchant')}
                  accessibilityRole={CONST.ACCESSIBILITY_ROLE.TEXT}
                  ref={(el) => (inputRef.current = el)}
+                 autoGrowHeight
              />
```

### What alternative solutions did you explore? (Optional)

**Comment by ayazalavi** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When user select Request Money from menu item then click on show more in the last step then tap on Merchant and then enter some text and insert emoji, this results in height getting fluctuated

### What is the root cause of that problem?
The root cause of the problem is the 'line-height' CSS attribute, which needs to be explicitly set for Safari. This is because, in Safari, the shadow DOM calculates the height of content manually. So, when we simply enter text, the shadow DOM calculates the height to be 17px. However, when we insert an emoji, the height is readjusted because there is room for an increase in height. After the emoji gets inserted, the height becomes 21px. [BaseTextInput](https://github.com/Expensify/App/blob/cbda2b03824d9b125ec6ab19e87ff533bea1b22f/src/components/TextInput/BaseTextInput.js#L330) has `lineHeight: undefined ` which is causing manual calculation in safari browser.

### What changes do you think we should make in order to solve the problem?
we need to change [BaseTextInput](https://github.com/Expensify/App/blob/cbda2b03824d9b125ec6ab19e87ff533bea1b22f/src/components/TextInput/BaseTextInput.js#L330) line to:
`!isMultiline && {height},`

which will take line height in consideration when passed through props and it is not causing any fluctuation anymore in safari. check video below for confirmation:
https://www.dropbox.com/scl/fi/kufyjdr5qupbw8rsezte8/Screen-Recording-2023-09-05-at-11.14.01-PM.mov?rlkey=oihzl2n3efd1yhpiocecvg1cj&dl=0

### What alternative solutions did you explore? (Optional)
Alternate solution is to check for browser and only do above changes where we see fluctuation in height.

**Comment by ayazalavi** [comment]
@sobitneupane ... TIA

**Comment by ayazalavi** [comment]
@sobitneupane any update regarding my proposal?

**Comment by sobitneupane** [comment]
@alphaboss1104 Thanks for the [proposal](https://github.com/Expensify/App/issues/26735#issuecomment-1705783835). 

`autoGrowHeight` is only used if for multiline input.

**Comment by sobitneupane** [comment]
Proposal from @ayazalavi looks good to me.

🎀 👀 🎀 C+ reviewed

**Comment by arosiclair** [comment]
@ayazalavi can you verify this doesn't cause any regressions wrt this comment:

> // Explicitly remove `lineHeight` from single line inputs so that long text doesn't disappear 
 // once it exceeds the input space (See https://github.com/Expensify/App/issues/13802)

**Comment by ayazalavi** [comment]
@arosiclair I can double checked https://github.com/Expensify/App/issues/13802. It is working as expected.

**Comment by arosiclair** [comment]
Be sure to include testing for #13802 in your PR test steps.

**Comment by ayazalavi** [comment]
@sobitneupane sharing PR in a while. waiting for offer.

**Comment by ayazalavi** [comment]
@arosiclair @sobitneupane @flaviadefaria Still waiting for my proposal to be accepted over the Upwork

**Comment by ayazalavi** [comment]
working on my PR.

**Comment by ayazalavi** [comment]
@sobitneupane PR ready: https://github.com/Expensify/App/pull/27480

**Comment by ayazalavi** [comment]
@arosiclair I havn't received offer over the upwork yet. Can you please assign this project to me on the upwork as well?

**Comment by ayazalavi** [comment]
@sobitneupane @arosiclair added new PR: https://github.com/Expensify/App/pull/28126

**Comment by ayazalavi** [comment]
@MelvinBot should I wait for upwork offer? Cant anyone manually send offer? @arosiclair @flaviadefaria @sobitneupane

**Comment by sobitneupane** [comment]
@ayazalavi Let's fix the regression.

**Comment by ayazalavi** [comment]
@sobitneupane PR made https://github.com/Expensify/App/pull/28580

**Comment by flaviadefaria** [comment]
Switching this to daily so that I can issue payment this week.

**Comment by ayazalavi** [comment]
@flaviadefaria can you share upwork offer with me so I can accept job?

**Comment by sobitneupane** [comment]
@flaviadefaria We had a regression from this PR. So, payment should be issued only after we resolve regression.

**Comment by flaviadefaria** [comment]
Sounds good, I'll wait until that is resolved before issuing payment.

**Comment by flaviadefaria** [comment]
Payment Summary:

>For reference, here are some details about the assignees on this issue:
@ayazalavi requires payment
@sobitneupane does not require payment (Eligible for Manual Requests)
@aman-atg requires payment [offer](https://www.upwork.com/nx/wm/offer/26633158) (Reporter)

- Merged after 6 business days - 50% penalty (days elapsed: 9)
- Regression - 50% penalty

@ayazalavi =  $500 - 50% (over 6 days) - 50% (regression) = $125
@sobitneupane  =  $500 - 50% (over 6 days) - 50% (regression) = $125
@aman-atg = $50 (reporter)

**Comment by flaviadefaria** [comment]
I'm issuing payment today.

**Comment by flaviadefaria** [comment]
@ayazalavi = can you share your UW profile link so that I can send you an offer?
@aman-atg = paid

**Comment by flaviadefaria** [comment]
>@flaviadefaria We had a regression from the PR. So, payment should be issued only after we resolve regression.
@sobitneupane, just to confirm has the regression been resolved?

**Comment by ayazalavi** [comment]
@flaviadefaria https://www.upwork.com/freelancers/~01a53213c1c66acb2d

**Comment by sobitneupane** [comment]
@flaviadefaria [PR](https://github.com/Expensify/App/pull/28580) got merged yesterday. We should still wait a week before issuing payment.

**Comment by flaviadefaria** [comment]
Payment Summary

@ayazalavi = $500 - 50% (over 6 days) - 50% (regression) = $125
@sobitneupane = $500 - 50% (over 6 days) - 50% (regression) = $125
@aman-atg = $50 (reporter)

@aman-atg = paid
@ayazalavi =  offer sent
@sobitneupane can you please fill out the comment above? Thanks!

**Comment by flaviadefaria** [comment]
@ayazalavi you've been paid. Closing this GH.

**Comment by sobitneupane** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [ ]  The PR that introduced the bug has been identified. Link to the PR:

https://github.com/Expensify/App/pull/14273
> * [ ] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

https://github.com/Expensify/App/pull/14273/files#r1375797245
> * [ ] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Not Required
> * [ ] Determine if we should create a regression test for this bug.

Yes
> * [ ] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Comment by sobitneupane** [comment]
https://github.com/Expensify/App/issues/26735#issuecomment-1774688328

Requested payment on newDot.

**Comment by JmillsExpensify** [comment]
$125 payment approved for @sobitneupane based on [this comment](https://github.com/Expensify/App/issues/26735#issuecomment-1774688328).
